src/library/scala/collection/immutable/Vector.scala

```diff
@@ -202,12 +202,12 @@ final class Vector[+A] private[immutable] (private[collection] val startIndex: I
     // Implementation similar to `appendAll`: when of the collections to concatenate (either `this` or `prefix`)
     // has a small number of elements compared to the other, then we add them using `:+` or `+:` in a loop
     import Vector.{Log2ConcatFaster, TinyAppendFaster}
-    if (prefix.isEmpty) this
+    if (prefix.iterator.isEmpty) this
     else {
-      prefix.size match {
+      prefix.iterator.size match {
```

Comment:
Maybe this changes fix https://github.com/scala/bug/issues/11122 by chance. But current implementation not efficient.

Reply:
revert this changes. create another pull request https://github.com/scala/scala/pull/7153